### PR TITLE
Add local Makefile for automated code updates

### PR DESCRIPTION
**Pull request by caseydavenport**
## Description

Adds a Makefile target to run which will mirror api code from the
source-of-truth in github.com/projectcalico/calico/api

For example:

```
make CALICO_VERSION=master update -f Makefile.local
```

Will submit a subsequent PR which actually runs the target. 

## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: Makefile.local [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# This is the local Makefile for the projectcalico/api repository, with targets
+# specific to github.com/projectcalico/api. This is opposed to Makefile, which
+# is mirrored from github.com/projectcalico/calico/api.
+
+# update pulls in the latest contents of this repository from the upstream
+# github.com/projectcalico/calico/api directory.
+CALICO_VERSION ?= $(shell git rev-parse --abbrev-ref HEAD)
+update:
+	rm -rf /tmp/calico-api-mirror
```
Comment (fasaxc) [inline Makefile.local:9]: Worth a check at the start that the repo isn't dirty?

File: Makefile.local [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+	mkdir -p /tmp/calico-api-mirror
+	git clone --depth 1 git@github.com:projectcalico/calico.git -b $(CALICO_VERSION) /tmp/calico-api-mirror
+	cp -r /tmp/calico-api-mirror/api/. .
```
Comment (fasaxc) [inline Makefile.local:12]: Doesn't handle the case of a file being deleted, I think.

**Review by fasaxc — CHANGES_REQUESTED** [review]

File: Makefile.local (hunk 2 above)
Comment (fasaxc) [reply Makefile.local:12]: I might have reached for `git read-tree` or something after pulling the monorepo as a remote.  But on reflection, why use an arcane incantation when `cp` will do :laughing:

**Comment by caseydavenport** [comment]
There's still a manual component to this, which is reviewing the changes that get pulled over and adding any to git tracking that need to be. But I think that's acceptable for now. We really just need something quick and dirty we can do for each minor release.

**Pull request merged**